### PR TITLE
Matrix cleanup 4.5

### PR DESCRIPTION
**Pull request by siefkenj**
This builds on #12390 by deprecating the determinant methods talked about in #12389 

Introduced `MatrixDeprecated` to house deprecated matrix
methods until they are ready for removal.  Methods that have
been deprecated are:

		berkowitz
		berkowitz_det
		berkowitz_eigenvals
		berkowitz_minors
		berkowitz_charpoly
		cofactorMatrix
		det_bareiss
		det_LU_decomposition
		minorEntry
		minorMatrix

They have been replaced with the methods

		det
		charpoly
		eigenvals
		cofactor_matrix
		minor
		minor_submatrix

### CODE REVIEW

**Comment by smichr** [comment]
I see the @moorepants merged "Phase 1". What is the next thing to merge and is there a preferred person to look at it and merge it?

**Comment by siefkenj** [comment]
Step 4 ( PR #12390 ) is the next thing to merge.  I don't know if there is a preferred person to merge it, but when things get merged, I can start Step 5, which would be separating out the eigen-vector-related operations and the row-reduction-based operations.

**Comment by asmeurer** [comment]
@smichr no one is preferred to merge it. If you feel these refactor PRs are OK to merge, feel free to do so. I don't want @siefkenj to get blocked for lack of reviewing.

**Comment by siefkenj** [comment]
@jksuom @smichr would you mind taking a look at this PR?

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -2040,7 +2044,154 @@ def minor_submatrix(self, i, j):
...
+
+        return berk + tuple(map(tuple, polys))
+
+    @deprecated(useinstead="cofactor_matrix", issue=12389, deprecated_since_version="1.1")
+    def cofactorMatrix(self, method="berkowitz"):
+        return self.cofactor_matrix(method=method)
+
+    @deprecated(useinstead="det(method='bareiss')", issue=12363, deprecated_since_version="1.1")
+    def det_bareis(self):
+        return self.det(method='bariess')
```
Comment (jksuom) [inline sympy/matrices/matrices.py]: bareiss

**Comment by jksuom** [comment]
Looks good to me (with typo fixed).

**Comment by jksuom** [comment]
This is ready. Thanks!

**Pull request merged**